File: src/lib/hitlToken.ts
import { invoke } from "@tauri-apps/api/core";

export type HitlScope = "shell_run" | "elevated";

export async function issueHitlApprovalToken(
  command: string,
  scope: HitlScope
): Promise<string> {
  return invoke<string>("issue_hitl_approval_token", { command, scope });
}

File: src-tauri/src/hitl_token.rs
pub use gnomad_core::agent::tokens::hitl::*;

#[tauri::command]
pub fn issue_hitl_approval_token(command: String, scope: String) -> Result<String, String> {
    let scope = HitlScope::parse(&scope).ok_or_else(|| {
        crate::error::into_invoke_err(crate::error::GnomadError::ShellValidation {
            message: format!("Invalid HITL scope: {scope}"),
            detail: Some("Use shell_run or elevated.".into()),
        })
    })?;
    issue_approval_token(&command, scope)
}
